pkg/cli/image/extract/extract.go

```diff
@@ -343,8 +346,14 @@ func (o *ExtractOptions) Run() error {
 					return fmt.Errorf("unable to connect to image repository %s: %v", from.String(), err)
 				}
 
+				isList := false
 				srcManifest, location, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
-				if err != nil {
+				if err == nil {
```

Comment:
Here similarly to above, `isList` is escaping its necessary scope by being defined outside the error check condition's body. You don't need that outside it. 
So I'd re-write it as:
```golang
srcManifest, location, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
if err != nil {
    .. // the 2 conditions ISImageForbidden & IsImageNotFound
    if _, ok := srcManifest.(*manifestlist.DeserializedManifestList); ok {
    .. // the body of if isList condition from below
```

Reply:
Also since you're using that two times already, I'd consider creating a function to encapsulate that logic.

Reply:
updated, fixed, didn't add the fn bc append doesn't include the 2 extra error checks, i'll add a fn if we should add the ImageForbidden/ImageNotFound checks though.